### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Romero10**
**Summary**: Response code is NOT 200 when getting a list of zip codes

**Preconditions**: I am authorized user.

**Steps:**
1. Send GET request to /zip-codes endpoint

**Actual Result**: Response code is 201.
**Expected Result**: Response code should be 200.

**Automation Test Coverage**
|| Test Class || Test Name ||
|  TC_ZipCode  |  availableZipCodesTest   |

--------------------------------------------------------------------------------
**Summary**: There are duplicates in available zip codes when adding a list of zip codes

**Preconditions**: I am authorized user.

**Steps**:
1. Send GET request to /zip-codes endpoint and select one of zip code
2. Send POST request to /zip-codes/expand endpoin with duplicate zip code (from step 1)
3. Send GET request to /zip-codes endpoint and make sure that no duplicate for zip code (from step 1)

**Actual Result**: There are duplicates in available zip codes.
**Expected Result**: There should be no duplicates in available zip codes.

**Automation Test Coverage**
|| Test Class || Test Name ||
|  TC_ZipCode  |  verifyAddDuplicationsZipCodesTest   |

--------------------------------------------------------------------------------
**Summary**: There are duplicates between available zip codes and already used zip codes

**Preconditions**: I am authorized user.

**Steps**:
1. Send GET request to /users endpoint and select one of zip code from any user
2. Send POST request to /zip-codes/expand endpoin with duplicate zip code (from step 1)
3. Send GET request to /zip-codes endpoint and make sure that zip code is NOT added (from step 1)

**Actual Result:** The already used zip code is added in available zip codes.
There are duplicates in available zip codes.
**Expected Result**: The already used zip code should NOT be added in available zip codes. 
There should be no duplicates in available zip codes (regarding to already used zip code).

**Automation Test Coverage**
|| Test Class || Test Name ||
|  TC_ZipCode  |  verifyAddDuplicationsUsedZipCodesTest   |

### CODE REVIEW

File: src/main/java/api/training/dto/UserDto.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class UserDto {
+
+	@JsonProperty("age")
```
Comment (EugenBorisik) [inline src/main/java/api/training/dto/UserDto.java:11]: Optional: if json field name equals to java field name, then @JsonProperty annotation is not required

File: src/main/java/api/training/services/UserService.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+	public static Pair<Integer, List<UserDto>> getUsers() {
+		Header headerToken = getTokenHeader(SCOPE_READ);
+
+		HttpUriRequest request = RequestBuilder.get()
+				.setUri(EndPoints.USERS)
+				.setHeader(headerToken)
+				.build();
+
+		CloseableHttpResponse response = client.request(request);
```
Comment (EugenBorisik) [inline src/main/java/api/training/services/UserService.java:31]: You need to close responses

File: src/test/java/TC_ZipCode.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import org.testng.internal.collections.Pair;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TC_ZipCode {
+
+	private final static String NAME = "Maxim";
+	private final static int AGE = 14;
+	private final static Sex SEX = Sex.MALE;
```
Comment (EugenBorisik) [inline src/test/java/TC_ZipCode.java]: Generate random test data, do not use static values

File: src/test/java/TC_ZipCode.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+	private SoftAssert softAssert;
+
+	@BeforeClass
+	public void createUser() {
+		List<UserDto> usersList = UserService.getUsers().second();
+
+		if (usersList.size() == 0) {
+			Pair<Integer, List<String>> availableZipCodes = ZipCodeService.getAvailableZipCodes();
+			randomUsedZipCode = availableZipCodes.second()
+					.get(RandomUtils.nextInt(0, availableZipCodes.second().size() - 1));
```
Comment (EugenBorisik) [inline src/test/java/TC_ZipCode.java:34]: What if there is no available zip codes?

File: src/test/java/TC_ZipCode.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+	@Test
+	public void verifyAddZipCodesTest() {
+		int responseStatusCode = ZipCodeService.addZipCodes(zipCodes);
+		softAssert.assertEquals(responseStatusCode, HttpStatus.SC_CREATED,
+				"Response code is NOT 201 when adding a list of zip codes.");
+
+		Pair<Integer, List<String>> availableZipCodes = ZipCodeService.getAvailableZipCodes();
+
+		zipCodes.forEach(zipCode -> softAssert.assertTrue(availableZipCodes.second().contains(zipCode),
+				"Zip Code '" + zipCode + "' are NOT added to available zip codes of application."));
```
Comment (EugenBorisik) [inline src/test/java/TC_ZipCode.java]: Optional: I recommend to use String.format instead of + .. + ...

**Review by EugenBorisik — CHANGES_REQUESTED** [review]

File: src/main/java/api/training/services/UserService.java (hunk 2 above)
Comment (Romero10) [reply src/main/java/api/training/services/UserService.java:31]: Added the ability to close responses after requests

File: src/test/java/TC_ZipCode.java (hunk 4 above)
Comment (Romero10) [reply src/test/java/TC_ZipCode.java:34]: Fixed.

File: src/test/java/TC_ZipCode.java (hunk 3 above)
Comment (Romero10) [reply src/test/java/TC_ZipCode.java]: Fixed.

**Review by EugenBorisik — APPROVED** [review]

**Pull request merged**